corehq/apps/app_manager/fields.py

```diff
@@ -420,14 +422,16 @@ def _sort_key_form(form):
     def _all_forms(self):
         forms = []
         unknown_forms = []
-        for f in get_exports_by_form(self.domain):
+
+        for f in get_exports_by_form(self.domain, use_es=self.domain_object.exports_use_elasticsearch):
             form = f['value']
             if form.get('app_deleted') and not form.get('submissions'):
                 continue
             if 'app' in form:
                 form['has_app'] = True
                 forms.append(form)
-            else:
+            elif not self.domain_object.exports_use_elasticsearch:
```

Comment:
Hi Farid, can you explain what does it mean when `app` is not in `form`? I would prefer not to exclude those forms, because when I run in a shell, your `_get_export_forms_by_app_es` 
 is already quick enough to solve the current issue. 500 error seems like another issue. Loading those either unknown or delete apps used to work without 500 error, at least until 2 months ago. If there is some changes that resulted in those 500 errors, we should fix that, instead of just exclude those forms.

Reply:
(Offlining a response to this)